bpfctl/src/main.rs

```diff
@@ -234,6 +269,115 @@ async fn main() -> anyhow::Result<()> {
+                        other => bail!("{} is not a valid direction", other),
+                    };
+                    let mut proc_on = Vec::new();
+                    if !proceed_on.is_empty() {
+                        for i in proceed_on.iter() {
+                            let action = ProceedOn::try_from(i.to_string())?;
+                            proc_on.push(action as i32);
+                        }
+                    }
+                    Some(load_request::AttachType::NetworkMultiAttach(
+                        NetworkMultiAttach {
+                            priority: *priority,
+                            iface: iface.to_string(),
+                            position: 0,
+                            direction: attach_direction as i32,
+                            proceed_on: proc_on,
+                        },
+                    ))
+                }
+                LoadCommands::Tracepoint { tracepoint } => {
+                    Some(load_request::AttachType::SingleAttach(SingleAttach {
+                        name: tracepoint.to_string(),
+                    }))
+                }
+            };
+
+            let image_pull_policy: bpfd_api::v1::ImagePullPolicy =
+                l.image_pull_policy.as_str().try_into()?;
+
+            let mut global_data: HashMap<String, Vec<u8>> = HashMap::new();
+
+            if let Some(global) = &l.global {
+                for g in global.iter() {
+                    global_data.insert(g.name.to_string(), g.value.clone());
+                }
+            }
+
+            let location = match l.registry_auth.clone() {
+                Some(a) => {
+                    let auth_raw = general_purpose::STANDARD_NO_PAD.decode(a)?;
+
+                    let auth_string = String::from_utf8(auth_raw)?;
+
+                    let (username, password) = auth_string.split(':').next_tuple().unwrap();
+
+                    load_request::Location::Image(BytecodeImage {
+                        url: l.image_url.clone(),
+                        image_pull_policy: image_pull_policy as i32,
+                        username: username.to_owned(),
+                        password: password.to_owned(),
+                    })
+                }
+                None => load_request::Location::Image(BytecodeImage {
+                    url: l.image_url.clone(),
+                    image_pull_policy: image_pull_policy as i32,
+                    username: "".to_owned(),
+                    password: "".to_owned(),
+                }),
+            };
+
+            let request = tonic::Request::new(LoadRequest {
+                location: Some(location),
+                section_name: l.section_name.to_string(),
+                program_type: prog_type as i32,
+                attach_type,
+                global_data,
+            });
+            let response = client.load(request).await?.into_inner();
+            println!("{}", response.id);
+        }
+
```

Comment:
This main is getting quite large. It might be helpful for future readers and for maintenance to compose some of the logic into discreet functions.

Reply:
#412